dipy/reconst/dti.py

```diff
@@ -1634,71 +1623,44 @@ def err_func(self, tensor, design_matrix, data, weighting=None, sigma=None):
+        else:
+            # Return the weighted residuals:
             with warnings.catch_warnings():
                 warnings.simplefilter("ignore")
-                w = sigma
-
-        # Return the weighted residuals:
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
-            self.sqrt_w = np.sqrt(w)
-            ans = self.sqrt_w * residuals
-            if np.iterable(w):
-                # cache the weights for the *non-squared* residuals
-                self.sqrt_w = np.sqrt(w)[:, None]
-            return ans
-
-    def jacobian_func(self, tensor, design_matrix, data, weighting=None, sigma=None):
-        """The Jacobian is the first derivative of the error function
-        :footcite:p:`Koay2006c`.
+                self.sqrt_w = np.sqrt(weights)
+                ans = self.sqrt_w * residuals
+                if np.iterable(weights):
+                    # cache the weights for the *non-squared* residuals
+                    self.sqrt_w = self.sqrt_w[:, None]
+                return ans
+
+    def jacobian_func(self, tensor, design_matrix, data, weights=None):
+        r"""The Jacobian is the first derivative of the error function [1]_.
 
         Parameters
         ----------
-        tensor : array (3, 3)
-            Hermitian matrix representing a diffusion tensor.
-        design_matrix : array (g, Npar)
-            Design matrix holding the covariants used to solve for the
-            regression coefficients. First six parameters of design matrix
-            should correspond to the six unique diffusion tensor elements in
-            the lower triangular order (Dxx, Dxy, Dyy, Dxz, Dyz, Dzz), while
-            last parameter to -log(S0)
-        data : array ([X, Y, Z, ...], g)
-            Data or response variables holding the data. Note that the last
-            dimension should contain the data. It makes no copies of data.
-        weighting: str, optional
-            the weighting scheme to use in considering the
-            squared-error. Default behavior is to use uniform weighting. Other
-            options: 'sigma' 'gmm'
-        sigma : array, optional
-            If 'sigma' weighting is used, we will weight the error function
-            according to 1/sigma^2. If 'gmm', the Geman-Mclure
-            M-estimator is used for weighting (see below).
+        tensor : array (3,3)
+            The 3-by-3 tensor matrix
+
+        design_matrix : array
+            The design matrix
+
+        data : array
+            The voxel signal in all gradient directions
+
+        weights : array ([X, Y, Z, ...], g), optional
+            Weights to apply for fitting. These weights must correspond to the
+            squared residuals such that $S = \sum_i w_i r_i^2$.
 
         Notes
         -----
-        This is an implementation of equation 14 in :footcite:p:`Koay2006c`.
+        This is an implementation of equation 14 in [1]_, but also
```

Comment:
please, keep the `:footcite:p:`

Reply:
Sorted, these references were left over anyway, not actually needed.